src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -20296,6 +20296,56 @@ struct NegDivConstSimplify final
   }
 };
 
+struct ReshapeBroadcastInDimSimplify final
+    : public CheckedOpRewritePattern<stablehlo::ReshapeOp,
+                                     ReshapeBroadcastInDimSimplify> {
+  using CheckedOpRewritePattern::CheckedOpRewritePattern;
+
+  LogicalResult matchAndRewriteImpl(stablehlo::ReshapeOp op,
+                                    PatternRewriter &rewriter) const {
+    auto bcastInDimOp =
+        op.getOperand().getDefiningOp<stablehlo::BroadcastInDimOp>();
+    if (!bcastInDimOp)
+      return failure();
+
+    if (!isOnlyUsedInOperation(bcastInDimOp, op))
```

Comment:
You should look at the reshape helper functions I made for all the reshapeX ops, which also is worth checking if subsumes this

Reply:
will take a look. I am getting these working first to see the effect on the FNO model. Just this set enables a perf gain of 3% over XLA for the primal

Reply:
Yeah fair nuff, in which case feel free to merge and maybe open an issue. The helper functions should handle both axes added and removed